server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1237,6 +1368,26 @@ public synchronized void updateFromClusterManager(
         assert invariant();
     }
 
+    /**
+     * Returns the replication mode considering the remote translog existence, current/primary/primary target allocation ids.
+     *
+     * @param allocationId              given allocation id
+     * @param primaryAllocationId       primary allocation id
+     * @param primaryTargetAllocationId primary target allocation id
+     * @return the replication mode.
+     */
+    private ReplicationMode getReplicationMode(String allocationId, String primaryAllocationId, String primaryTargetAllocationId) {
+        // If remote translog is enabled, then returns replication mode checking current allocation id against the
+        // primary and primary target allocation id.
+        if (indexSettings().isRemoteTranslogStoreEnabled()) {
+            return (allocationId.equals(primaryAllocationId) || allocationId.equals(primaryTargetAllocationId))
+                ? ReplicationMode.FULL_REPLICATION
+                : ReplicationMode.NO_REPLICATION;
+        }
+        // For all other case, it is logical replication mode.
+        return ReplicationMode.FULL_REPLICATION;
+    }
+
```

Comment:
I see `ReplicationTracker` as a means to manage state, so it shouldn't dictate how a replication should be performed which should well be the task of `ReplicationProxy` which is also what it is partially performing.
`ReplicationTracker` however could have a state it needs for all its tracking and state management responsibilities

Reply:
@Bukhtawar  incorporated